src/Google.Storage.V1/Preconditions.cs

```diff
@@ -23,151 +24,13 @@
-        {
-            if (!condition)
-            {
-                throw new InvalidOperationException(string.Format(format, arg0, arg1));
-            }
-        }
-
-
-        /// <summary>
-        /// Checks that given argument-based condition is met, throwing an <see cref="ArgumentException"/> otherwise.
-        /// </summary>
-        /// <param name="condition">The (already evaluated) condition to check.</param>
-        /// <param name="paramName">The name of the parameter whose value is being tested.</param>
-        /// <param name="message">The message to include in the exception, if generated. This should not
-        /// use interpolation, as the interpolation would be performed regardless of whether or not an exception
-        /// is thrown.</param>
-        internal static void CheckArgument(bool condition, string paramName, string message)
-        {
-            if (!condition)
-            {
-                throw new ArgumentException(message, paramName);
-            }
-        }
-
-        /// <summary>
-        /// Checks that given argument-based condition is met, throwing an <see cref="ArgumentException"/> otherwise.
-        /// </summary>
-        /// <param name="condition">The (already evaluated) condition to check.</param>
-        /// <param name="paramName">The name of the parameter whose value is being tested.</param>
-        /// <param name="format">The format string to use to create the exception message if the
-        /// condition is not met.</param>
-        /// <param name="arg0">The argument to the format string.</param>
-        internal static void CheckArgument<T>(bool condition, string paramName, string format, T arg0)
-        {
-            if (!condition)
-            {
-                throw new ArgumentException(string.Format(format, arg0), paramName);
-            }
-        }
-
-        /// <summary>
-        /// Checks that given argument-based condition is met, throwing an <see cref="ArgumentException"/> otherwise.
-        /// </summary>
-        /// <param name="condition">The (already evaluated) condition to check.</param>
-        /// <param name="paramName">The name of the parameter whose value is being tested.</param>
-        /// <param name="format">The format string to use to create the exception message if the
-        /// condition is not met.</param>
-        /// <param name="arg0">The first argument to the format string.</param>
-        /// <param name="arg1">The second argument to the format string.</param>
-        internal static void CheckArgument<T1, T2>(bool condition, string paramName, string format, T1 arg0, T2 arg1)
-        {
-            if (!condition)
-            {
-                throw new ArgumentException(string.Format(format, arg0, arg1), paramName);
-            }
-        }
-
         internal static T CheckEnumValue<T>(T value, string paramName) where T : struct
```

Comment:
Why not put this in GAX too? It'd be nice to get all preconditions out of these individual libraries. I agree this particular method is slightly specialist, but I don't think that matters.